src/mds/MDSMap.cc

```diff
@@ -702,30 +637,40 @@ void MDSMap::encode(bufferlist& bl, uint64_t features) const
   encode(data_pools, bl);
   encode(cas_pool, bl);
 
-  __u16 ev = 15;
+  __u16 ev = 5;
   encode(ev, bl);
-  encode(compat, bl);
-  encode(metadata_pool, bl);
+  {
+    CompatSet compat;
+    encode(compat, bl);
+  }
+  encode((int64_t)0, bl);
   encode(created, bl);
   encode(modified, bl);
-  encode(tableserver, bl);
+  encode((int32_t)0, bl);
   encode(in, bl);
   encode(inc, bl);
   encode(up, bl);
   encode(failed, bl);
   encode(stopped, bl);
   encode(last_failure_osd_epoch, bl);
-  encode(ever_allowed_features, bl);
-  encode(explicitly_allowed_features, bl);
-  encode(inline_data_enabled, bl);
-  encode(enabled, bl);
-  encode(fs_name, bl);
-  encode(damaged, bl);
-  encode(balancer, bl);
-  encode(standby_count_wanted, bl);
-  encode(old_max_mds, bl);
-  encode(min_compat_client, bl);
   ENCODE_FINISH(bl);
+
+  {
+    // ev part
+    ENCODE_START(1, 1, bl);
+    encode(metadata_pool, bl);
+    encode(tableserver, bl);
+    encode(ever_allowed_features, bl);
+    encode(explicitly_allowed_features, bl);
+    encode(inline_data_enabled, bl);
+    encode(enabled, bl);
+    encode(damaged, bl);
+    encode(balancer, bl);
+    encode(standby_count_wanted, bl);
+    encode(old_max_mds, bl);
+    encode(min_compat_client, bl);
```

Comment:
inline_data_enabled, enabled, damaged should belong to normal part.  the ev part encode up to damaged